File: BACKEND/facturador_backend/src/main/java/com/bap/api/services/api/ApiDosificacionService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.bap.api.services.api;

import com.bap.api.configurate.AbstractEntity;
import com.bap.api.dto.Respuesta;
import com.bap.api.dto.SolicitudCliente;
import com.bap.api.model.api.ApiDosificacion;

/**
 *
 * @author ruth
 */
public interface ApiDosificacionService extends AbstractEntity<ApiDosificacion> {

    Respuesta solicitudCuis(SolicitudCliente t);
    
    Respuesta cierreOperaciones(SolicitudCliente t);

    ApiDosificacion getDosificacionVigenteBySucursal(Long idSucursal);

    ApiDosificacion getDosificacionVigenteByPuntoVenta(Long idPuntoVenta);
}

File: FRONTEND/facturador-frontend/src/services/ApiDosificacionService.js
import Api from '@/services/Api'

export default {
    ///ApiDosificacion
    solicitudCuis(dato){
        return Api().post("/api/apiDosificacion/solicitudCuis", dato);
    },
    cierreOperacion(dato){
        return Api().post("/api/apiDosificacion/cierreOperacion", dato);
    },
    dosificacionVigente(login){
        return Api().get(`/api/apiDosificacion/getDosificacionVigente/${login}`);
    },
    getDosificacionVigenteBySucursal(idSucursal){
        return Api().get(`/api/apiDosificacion/getDosificacionVigenteBySucursal/${idSucursal}`);
    },
    getDosificacionVigenteByPuntoVenta(idPuntoVenta){
        return Api().get(`/api/apiDosificacion/getDosificacionVigenteByPuntoVenta/${idPuntoVenta}`);
    },
    listaDosificacionSucursalPorIdEmpresa(idEmpresa){
        return Api().get(`/api/apiDosificacion/listaDosificacionSucursalPorIdEmpresa/${idEmpresa}`);
    },
    listaDosificacionPuntoVentaPorIdEmpresa(idEmpresa){
        return Api().get(`/api/apiDosificacion/listaDosificacionPuntoVentaPorIdEmpresa/${idEmpresa}`);
    },
}
